cluster-autoscaler/FAQ.md

```diff
@@ -297,140 +308,143 @@ kubectl annotate node <nodename> cluster-autoscaler.kubernetes.io/scale-down-dis
-or pods created by daemonsets) can be moved to some other nodes. Stand-alone pods which are not
-under control of a deployment, replica set, replication controller or job would not be recreated
-if the node is deleted so they make a node needed, even if its utilization is low. While
-checking this condition the new locations of all pods are memorized. With that Cluster Autoscaler
-knows where each pod can be moved and which nodes depend on which other nodes in terms of
-pod migration. Of course, it may happen that eventually the scheduler will place the pods
-somewhere else.
-
-* All system pods running on the node (except these that run on all nodes by default like
-manifest-run pods or pods created by daemonsets) have a PodDisruptionBudget.
-
-* There are no pods with local storage. Applications with local storage would lose their
-data if a node is deleted, even if they are replicated.
-
-* It doesn't have scale down disabled annotation (see [How can I prevent Cluster Autoscaler from scaling down a particular node?](#how-can-i-prevent-cluster-autoscaler-from-scaling-down-a-particular-node))
-
-If a node is not needed for more than 10 min (configurable) then it can be deleted. Cluster Autoscaler
-deletes one node at a time to reduce the risk of creating new unschedulable pods. The next node
-can be deleted when it is also not needed for more than 10 min. It may happen just after
-the previous node is fully deleted or after some longer time.
-
-What happens when a node is deleted? As mentioned above, all pods should be migrated elsewhere.
-For example if node A is deleted then its pods, consuming 400m CPU, are moved to, let's say, node
-X where is 450m CPU available. Ok, but what other nodes that also were eligible for deletion? Well,
-it depends. If node B also wanted to move its pods, consuming 350m CPU, to node X then it cannot
-do it anymore as there is almost no capacity left. It has to them somewhere else, and it is not sure that
-if A had been deleted much earlier than B, during the last 10 min, would always have a place to
-move its pods. So the requirement of being unused for 10 min may not be valid anymore for B.
-But if another node C, in case of deletion, can move its pods to node Y then it
-may still do it, because no one touched Y. So C can be deleted immediately after A, but B may not be
-deleted immediately.
+different strategies can be selected for choosing which node group is increased. Check [What are Expanders?](#what-are-expanders) section to learn more about strategies.
+
+It may take some time before the created nodes appear in Kubernetes. It almost entirely
+depends on the cloud provider and the speed of node provisioning. Cluster
+Autoscaler expects requested nodes to appear within 15 minutes
+(configured by `--max-node-provision-time` flag.) After this time, if they don't
+appear, it stops considering them in simulations and may attempt to scale up a
+different group if the pods are still pending.
```

Comment:
If we're going into that level of detail it may be worth mentioning that CA will actually delete unregistered nodes, not just stop considering them in simulations. This uses a separate flag `--unregistered-node-removal-time` and IIRC it's a bad idea to set `--unregistered-node-removal-time` and `--max-node-provision-time` to a different value.

Reply:
Working on it. BTW, do we have a use case for setting those flags to different values? If not, let's open an issue for changing it to one flag.

Reply:
We don't, +1 for making it a single flag.